packages/alfa-rules/src/sia-r113/rule.ts

```diff
@@ -0,0 +1,145 @@
+import { Page } from "@siteimprove/alfa-web";
+
+import { expectation } from "../common/act/expectation";
+
+import { targetsOfPointerEvents } from "../common/applicability/targets-of-pointer-events";
+
+import { WithBoundingBox, WithName } from "../common/diagnostic";
+
+import { hasSufficientSize } from "../common/predicate/has-sufficient-size";
+import { isUserAgentControlled } from "../common/predicate/is-user-agent-controlled";
+
+const { or } = Predicate;
+
+export default Rule.Atomic.of<Page, Element>({
+  uri: "https://alfa.siteimprove.com/rules/sia-r113",
+  requirements: [Criterion.of("2.5.8")],
+  evaluate({ device, document }) {
+    return {
+      applicability() {
+        return targetsOfPointerEvents(document, device);
+      },
+
+      expectations(target) {
+        // Existence of a bounding box is guaranteed by applicability
+        const box = target.getBoundingBox(device).getUnsafe();
+        const name = WithName.getName(target, device).getOr("");
+        return {
+          1: expectation(
+            isUserAgentControlled()(target),
+            () => Outcomes.IsUserAgentControlled(name, box),
+            hasSufficientSizeOrSpacing(document, device)(target)
+              ? () => Outcomes.HasSufficientSizeOrSpacing(name, box)
+              : () => Outcomes.HasInsufficientSizeAndSpacing(name, box),
+          ),
+        };
+      },
+    };
+  },
+});
+
+/**
+ * @public
+ */
+export namespace Outcomes {
+  export const HasSufficientSizeOrSpacing = (name: string, box: Rectangle) =>
+    Ok.of(
+      WithBoundingBox.of("Target has sufficient size or spacing", name, box),
+    );
+
+  export const HasInsufficientSizeAndSpacing = (name: string, box: Rectangle) =>
+    Err.of(
+      WithBoundingBox.of("Target has insufficient size and spacing", name, box),
+    );
+
+  export const IsUserAgentControlled = (name: string, box: Rectangle) =>
+    Ok.of(WithBoundingBox.of("Target is user agent controlled", name, box));
+}
+
+function hasSufficientSizeOrSpacing(
+  document: Document,
+  device: Device,
+): Predicate<Element> {
+  return or(
+    hasSufficientSize(24, device),
+    hasSufficientSpacing(document, device),
+  );
+}
+
+const spacingCache = Cache.empty<
+  Document,
+  Cache<Device, Cache<Element, boolean>>
+>();
+
+/**
+ * @remarks
+ * Spacing is calculated by
+ * 1. drawing a 24px diameter circle around the center of the bounding box of the target,
+ * 2. checking if the circle intersects with the bounding box of any other target, or
+ * 3. if the circle intersects with the 24px diameter circle of another undersized target.
+ */
+function hasSufficientSpacing(
+  document: Document,
+  device: Device,
+): Predicate<Element> {
+  return (target) => {
+    // The target might have been cached while computing the spacing for another target
+    const cachedTarget = spacingCache
+      .get(document, Cache.empty)
+      .get(device, Cache.empty)
+      .get(target);
```

Comment:
We need to do all the computation inside the `ifMissing` bit of the last `.get` to ensure the result is indeed cached.